apps/server/src/modules/room-membership/service/room-membership.service.ts

```diff
@@ -177,6 +193,80 @@ export class RoomMembershipService {
 		}
 	}
 
+	private async getStats(groupsOnPage: Group[], schoolId: string): Promise<RoomMembershipStats[]> {
+		const groupIds = groupsOnPage.map((group) => group.id);
+		const roomMemberships = await this.roomMembershipRepo.findByGroupIds(groupIds);
+		const groupIdOwnerMap = await this.getOwnerMap(groupsOnPage);
+
+		const stats = await this.getRoomMemberstatsForGroups(schoolId, groupsOnPage);
+
+		const result = roomMemberships.map((item) => {
+			const { userGroupId, schoolId, roomId } = item;
+			const stat = stats.get(userGroupId);
+			const owner = groupIdOwnerMap.get(userGroupId) ?? '';
+			return {
+				roomId: roomId,
+				roomSchoolId: schoolId,
+				owner,
+				totalMembers: 0,
+				internalMembers: 0,
+				externalMembers: 0,
+				...stat,
+			};
+		});
+		return result;
+	}
+
+	private async getOwnerMap(groupsOnPage: Group[]): Promise<Map<EntityId, string>> {
+		const onwerRole = await this.roleService.findByName(RoleName.ROOMOWNER);
+		const owners = groupsOnPage.map((group) => {
+			const owner = group.users.find((user) => user.roleId === onwerRole.id);
+			return {
+				groupId: group.id,
+				ownerUserId: owner?.userId,
+			};
+		});
+		const ownerUserIds = owners.map((owner) => owner.ownerUserId).filter((id): id is EntityId => id !== undefined);
+		const ownerUsers = await this.userService.findByIds(ownerUserIds);
+		const groupIdOwnerMap = new Map(
+			owners.map(({ groupId, ownerUserId }) => {
+				const ownerUser = ownerUsers.find((user) => user.id === ownerUserId);
+				const name = `${ownerUser?.firstName ?? ''} ${ownerUser?.lastName ?? ''}`.trim();
+				return [groupId, name];
+			})
+		);
+		return groupIdOwnerMap;
+	}
+	private async getRoomMemberstatsForGroups<T extends Group>(
```

Comment:
as the type is called `MemberStats` I would also follow this here with `...MemberStatsForGroups`

Reply:
changed.